frontend/src/app/MindMap.js

```diff
@@ -269,18 +273,39 @@ class MindMap extends React.Component {
         if (this.props.curProject === null || (this.state.loading && this.state.sources === null)) {
             return <Loader size="lg" backdrop center/>
         }
-
+        let newSourceForm;
+        if (this.state.newSourceFormType === "URL") {
+            newSourceForm =
+                <NewSourceForm showNewSourceForm={this.state.showNewSourceForm}
+                    newSourceData={this.state.newSourceData}
+                    curProject={this.props.curProject}
+                    renderNetwork={this.renderNetwork}
+                    inputType={"URL"}
+                    header = {"URL of the source"}
+                    placeholder = {"New Source URL"}
+                    inputComponentClass={"input"}
+                    switchShowNewSourceForm={this.switchShowNewSourceForm}/>;
+        } else if (this.state.newSourceFormType === "Notes") {
+            newSourceForm = 
+                <NewSourceForm showNewSourceForm={this.state.showNewSourceForm}
+                    newSourceData={this.state.newSourceData}
+                    curProject={this.props.curProject}
+                    renderNetwork={this.renderNetwork}
+                    inputType={"Note"}
+                    header = {"Note of the source"}
+                    placeholder = {"New Note"}
+                    inputComponentClass={"textarea"}
+                    switchShowNewSourceForm={this.switchShowNewSourceForm}/>
+        } else {
+            newSourceForm = null;
+        }
         return (
             <div>
                 <div id="mindmap"/>
                 <SourceView selectedNode={this.state.selectedNode}
                             setSelectedNode={this.setSelectedNode}
                             renderNetwork={this.renderNetwork}/>
-                <NewSourceForm showNewSourceForm={this.state.showNewSourceForm}
-                               newSourceData={this.state.newSourceData}
-                               curProject={this.props.curProject}
-                               renderNetwork={this.renderNetwork}
-                               switchShowNewSourceForm={this.switchShowNewSourceForm}/>
+                {newSourceForm}
```

Comment:
This is just a suggestion, but to make the `render()` function more compact you could create a separate function inside this class and put the code to create the new source form inside that, then you just call the function here. Check the implementation of `renderNewNotesButton()` (and other buttons) inside the `NotesList` class, which is in `SourceView.js`.